src/vehicle.cpp

```diff
@@ -2274,6 +2275,7 @@ void Vehicle::LeaveStation()
 
 	HideFillingPercent(&this->fill_percent_te_id);
 	trip_occupancy = CalcPercentVehicleFilled(this, nullptr);
+	this->last_station_departure_date = _date;
```

Comment:
date_of_last_service is applied to all vehicles in a consist (see VehicleServiceInDepot), which is on the same level as the question above:
What to return when a train has visited a station before, and is then rearranged in the depot to make a different engine the front engine.

Reply:
Rearranging should return same value as when newly built.